examples/transmission_main_example.py

```diff
@@ -77,10 +61,63 @@ def main(args):
     axis('off')
     show()
 
+
+def main(args):
+    input_filename = str(args.filename)
+    split_filename = input_filename.split(".")
+    json_filename = split_filename[0]+'.json'
+    try:
+        assert split_filename[1] in ('json','xls','csv','xlsm')
+    except AssertionError as e:
+        print(f'invalid file extension .{split_filename[1]}')
+        raise e
+    if split_filename[1] != 'json':
+        print(f'parse excel input to {json_filename}')
+        convert_file(input_filename)
+
+    json_data = load_json(json_filename)
+    read_eqpt_library(EQPT_LIBRARY_FILENAME)
+
+    network = network_from_json(json_data)
+    build_network(network)
+
+    spacing = 0.05 #THz
+    si = SpectralInformation() # !! SI units W, Hz
+    si = si.update(carriers=tuple(Channel(f, (191.3+spacing*f)*1e12, 
+            32e9, 0.15, Power(1e-3, 0, 0)) for f in range(1,97)))
+
+    trx = [n for n in network.nodes() if isinstance(n, Transceiver)]
+    if args.list>=1:
+        print(*[el.uid for el in trx], sep='\n')
+    else:
+        try:
+            source = next(el for el in trx if el.uid == args.source)
+        except StopIteration as e:
+            source = trx[0]
+            print(f'invalid souce node specified: {args.source!r}, replaced with {source}')
+
+        try:
+            sink = next(el for el in trx if el.uid == args.sink)
+        except StopIteration as e:
+            sink = trx[1]
+            print(f'invalid destination node specified: {args.sink!r}, replaced with {sink}')
+
+        path = dijkstra_path(network, source, sink)
+        print(f'There are {len(path)} network elements between {source} and {sink}')
+
+        for el in path:
+            si = el(si)
+            print(el)
+
+        #plot_network_graph(network, path, source, sink)
+
 parser = ArgumentParser()
 parser.add_argument('filename', nargs='?', type=Path,
-  default= Path(__file__).parent / 'edfa/edfa_example_network.json')
+  default= Path(__file__).parent / 'edfa_example_network.json')
+parser.add_argument('source', type=str, nargs='?', default="", help='source node')
+parser.add_argument('sink', type=str, nargs='?', default="", help='sink node')
 parser.add_argument('-v', '--verbose', action='count')
+parser.add_argument('-l', '--list', action='count', default=0, help='list all network nodes')
```

Comment:
This should be `action='store_true'` rather than `action='count'`.

Can you call this parameter "list-nodes" to make it clear what it is listing?

You can check if the parameter was passed by simply
`
if args.list_nodes:
  ...
`

Reply:
sorry, not a parser pro, even though I read the doc.
Please implement and I can learn from it ;)